### ISSUE TITLE
Problems with Games on Win95 on dynamic_x86 core

### ISSUE THREAD

**Issue by DSN9990** [joncampbell123/dosbox-x#2408]
Hello again. I have been testing Win95 with the recent changes, and of course, some games.
I have successfully run Hexen II on my Win95 OSR2 setup but another great classic game, called Daytona USA did not work at all with the dynamic_x86 core, it seems that the system hangs the moment you run the game with just a menu bar at the top (only the music is still heard) and I must restart the system using the Ctrl-Alt-Del option from the Send special key submenu (under Main). If I use normal core instead, the game seems to work fine (the intro appears), I guess, but it runs too slowly compared to the dynx86 core. Maybe there is a problem with dynamic_x86 core? Should this can be fixed? I really appreciate any help. Thanks!

**Comment by DSN9990** [comment]
@Wengier I tested Tomb Raider II and it also did not work at all unless I use the normal core. If I use normal core the game will run fine (but very slow). If I use dynamic_x86 core the game crashes at startup with a "ClearRenderBuffer" error, though. I noticed the dynamic_x86 core is still very bad with Win95, so would be good to fix these issues anyway.

**Comment by DSN9990** [comment]
Please reply to me when you have time.

**Comment by Wengier** [comment]
Windows 95/98 has become compatible with the dynamic_x86 core as a whole since DOSBox-X 0.83.10 (thanks to @koolkdev's work in PR #2182). But as you mentioned there can still be some remaining issues between Windows 9x and the dynamic_x86 core. @koolkdev may want to take a look at this when he has time.

**Comment by DSN9990** [comment]
@Wengier Thanks. I will talk about these issues when @koolkdev have time.

**Comment by joncampbell123** [comment]
There is a known issue with Windows 95, dynamic core, and the DOS "box" within Windows 95, or anything involving that level of virtual 8086 mode. It may have something to do with INT 10h, though more checking is needed to verify that.

**Comment by DSN9990** [comment]
@Wengier Well. I tried again Tomb Raider II using the latest build you compiled and dosbox-x crashes immediately when running it, Windows prompts me with a "DOSBox-X has stopped working" error.

**Comment by Wengier** [comment]
@DSN9990 Which build did you use to run the game (I uploaded multiple Windows binaries recently)? Have you tried the Visual Studio build below too?

* [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)

**Comment by DSN9990** [comment]
@Wengier I tried the MinGW x64 SDL1 build you compiled in #2410.

**Comment by Wengier** [comment]
@DSN9990 I believe it is likely the same problem as issue #2426. Can you try the updated MinGW binaries below?

64-bit SDL1: * [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

64-bit SDL2:  * [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by DSN9990** [comment]
@Wengier It still crashes anyway, Windows prompted me again with a "stopped working" error message, so please give me a VS2019 x64 SDL1 build. Because of this, I think this is a issue with MinGW.

**Comment by Wengier** [comment]
@DSN9990 The 64-bit Visual Studio SDL1 build is available below:

* [dosbox-x-64-update.zip](http://cndos.fam.cx/dosware/dosbox-x-64-update.zip)

I tested Tomb Raider II (demo version [here](https://tombraiders.net/stella/files/gamefiles/TR2files/tr2_pc_demo1.exe)), which appears to run with it. You can also check out your config file if it does not work.

**Comment by Wengier** [comment]
@DSN9990 An updated MinGW 64-bit SDL2 build is available from:

* [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by Wengier** [comment]
@DSN9990 An updated MinGW 64-bit SDL1 build is available from:

* [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

**Comment by DSN9990** [comment]
@Wengier I'm still having issues in Win95, most of the games freeze within a black screen unless I use normal core (although I can still hear the sound). Which build you think better? Win95 is almost unstable for me.

**Comment by DSN9990** [comment]
For example, I tried Scorched Planet in it with the dynamic_x86 core and it goes in a black screen as soon as I start the game, however I'm still able to hear sound and press a key. If I select normal core the game (the intro movies, main menu and the rest) will appear and render properly. Can someone reproduce this issue?

**Comment by DSN9990** [comment]
@Wengier Give me a build that work the best as possible which do not have the black screen issues I said above. Can you try to reproduce this on all of the builds above to surely confirm this?

**Comment by Wengier** [comment]
@DSN9990 I do not have a copy of Scorched Planet for Windows 95 at this time and I cannot really tell with full confidence which build will work the best. But it is possible that the issue you described is related to the Pentium MMX emulation. Can you first try the following build below, and if the issue persists, change the cputype to "pentium" or "auto" instead of "pentium_mmx" and see if there is any change?

* [dosbox-x-64-update.zip](http://cndos.fam.cx/dosware/dosbox-x-64-update.zip)

**Comment by DSN9990** [comment]
@Wengier I think the VS2019 x86 SDL1 build may be better for now. At least until these problems are solved.

**Comment by Wengier** [comment]
@DSN9990 Good to hear that. The updated MinGW builds are now also available from:

64-bit SDL1: [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

64-bit SDL2: [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by DSN9990** [comment]
@Wengier It seems that Scorched Planet for Win95 (and other games that did not work before because of the black screen) now work as expected, at least for me. I'm using the VS2019 x64 SDL1 build you provided above. So there are likely some issues between certain builds and the dynx86 core too.

**Comment by krixkrax** [comment]
hello, recently i'm messing with earthsiege 2 and i bumped a cd-rom problem during campaign.
![es2](https://user-images.githubusercontent.com/81972013/127739167-a9d1f411-790e-4634-ba7a-7a642ff40e8f.JPG)
of course the cd image file is mounted correctly, otherwise the game wouldn't start at all...
anyone experienced such problem before?
(winxp host, w95b osr 2 with mingw sdl2 0.83.15.)

**Comment by rderooy** [comment]
Your mounting it with ``imgmount``? Looking online, I found that the game used CD Audio, so you should have a CUE/BIN file, and not an ISO file.

In any case, perhaps the CD was not properly ripped, or the CUE file does not correctly reflect the real CD.

**Comment by krixkrax** [comment]
@rderooy 
yes, im using imgmount command. i tried with 3 different image files, cue/bin, ccd, iso... all the same. up to that point, the game runs fine with music without problem.
Im sure this bug relates to win95 somehow.

**Comment by krixkrax** [comment]
hola!
months ago i installed and played a space sim named "Terminus" on Win95B with glide emulation. the game ran perfectly in fullscreen, using nglide on winxp 32bit host.
recently i installed it again and i simply cant make it to work with the same method. 
i put the 68kb glide2x.dll in the game directory, i set glide=true, yet it wont start.
of course the game runs in low level emulation too, but only in windowed mode. (the sofware emulation is unplayable slow.)
i wanna play it in fullscreen again.

![t1](https://user-images.githubusercontent.com/81972013/129475765-bc563fda-5d1b-40c2-8b3b-3d006c842b59.JPG)
![t2](https://user-images.githubusercontent.com/81972013/129475766-baafdf39-8ff1-43c5-bbc0-1d19cce1abc3.JPG)

i think i didnt modified anything else in the dosbox.conf back then... if i did, i sadly dont remember ><
any help?
